openedx/core/djangoapps/video_pipeline/tests/test_api.py

```diff
@@ -0,0 +1,98 @@
+"""
+Tests for Video Pipeline api utils.
+"""
+import ddt
+from mock import Mock, patch
+
+from django.test.testcases import TestCase
+from slumber.exceptions import HttpClientError
+
+from student.tests.factories import UserFactory
+
+from openedx.core.djangoapps.video_pipeline.api import update_3rd_party_transcription_service_credentials
+from openedx.core.djangoapps.video_pipeline.tests.mixins import VideoPipelineIntegrationMixin
+
+
+@ddt.ddt
+class TestAPIUtils(VideoPipelineIntegrationMixin, TestCase):
+    """
+    Tests for API Utils.
+    """
+    def setUp(self):
+        self.pipeline_integration = self.create_video_pipeline_integration()
+        self.user = UserFactory(username=self.pipeline_integration.service_username)
+
+    def test_update_transcription_service_credentials_with_integration_disabled(self):
+        """
+        Test updating the credentials when service integration is disabled.
+        """
+        self.pipeline_integration.enabled = False
+        self.pipeline_integration.save()
+        is_updated = update_3rd_party_transcription_service_credentials()
+        self.assertFalse(is_updated)
+
+    def test_update_transcription_service_credentials_with_unknown_user(self):
+        """
+        Test updating the credentials when expected service user is not registered.
+        """
+        self.pipeline_integration.service_username = 'non_existent_user'
+        self.pipeline_integration.save()
+        is_updated = update_3rd_party_transcription_service_credentials()
+        self.assertFalse(is_updated)
+
+    @ddt.data(
+        (
+            {
+                'username': 'Jason_cielo_24',
+                'api_key': '12345678',
+            },
+            Mock(),
+            True
+        ),
+        (
+            {
+                'api_key': '12345678',
+                'api_secret': '11111111',
+            },
+            Mock(),
+            True
+        ),
+    )
+    @ddt.unpack
+    @patch('openedx.core.djangoapps.video_pipeline.api.log')
+    @patch('openedx.core.djangoapps.video_pipeline.utils.EdxRestApiClient')
+    def test_update_transcription_service_credentials(self, credentials_payload, pipeline_response,
+                                                      expected_is_updated, mock_client, mock_logger):
+        """
+        Tests that the update transcription service credentials api util works as expected.
+        """
+        # Mock the post request
+        mock_credentials_endpoint = mock_client.return_value.transcript_credentials
+        mock_credentials_endpoint.post = pipeline_response
+        # try updating the transcription service credentials
+        is_updated = update_3rd_party_transcription_service_credentials(**credentials_payload)
+
+        mock_credentials_endpoint.post.assert_called_with(credentials_payload)
+        self.assertEqual(mock_logger.exception.called, not expected_is_updated)
```

Comment:
I don't understand what are trying to do here? a comment here would be helpful too

Reply:
making sure `log.exception` wasn't called.